solidity/random-beacon/contracts/libraries/GasStation.sol

```diff
@@ -0,0 +1,57 @@
+// SPDX-License-Identifier: MIT
+//
+// ▓▓▌ ▓▓ ▐▓▓ ▓▓▓▓▓▓▓▓▓▓▌▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▄
+// ▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▌▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓
+//   ▓▓▓▓▓▓    ▓▓▓▓▓▓▓▀    ▐▓▓▓▓▓▓    ▐▓▓▓▓▓   ▓▓▓▓▓▓     ▓▓▓▓▓   ▐▓▓▓▓▓▌   ▐▓▓▓▓▓▓
+//   ▓▓▓▓▓▓▄▄▓▓▓▓▓▓▓▀      ▐▓▓▓▓▓▓▄▄▄▄         ▓▓▓▓▓▓▄▄▄▄         ▐▓▓▓▓▓▌   ▐▓▓▓▓▓▓
+//   ▓▓▓▓▓▓▓▓▓▓▓▓▓▀        ▐▓▓▓▓▓▓▓▓▓▓         ▓▓▓▓▓▓▓▓▓▓         ▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓
+//   ▓▓▓▓▓▓▀▀▓▓▓▓▓▓▄       ▐▓▓▓▓▓▓▀▀▀▀         ▓▓▓▓▓▓▀▀▀▀         ▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▀
+//   ▓▓▓▓▓▓   ▀▓▓▓▓▓▓▄     ▐▓▓▓▓▓▓     ▓▓▓▓▓   ▓▓▓▓▓▓     ▓▓▓▓▓   ▐▓▓▓▓▓▌
+// ▓▓▓▓▓▓▓▓▓▓ █▓▓▓▓▓▓▓▓▓ ▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓  ▓▓▓▓▓▓▓▓▓▓
+// ▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓ ▐▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓ ▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓▓  ▓▓▓▓▓▓▓▓▓▓
+//
+//                           Trust math, not hardware.
+
+pragma solidity ^0.8.6;
+
+library GasStation {
+    struct Data {
+        // Map holding info about all deposits: address => (slot => value).
+        mapping(address => mapping(uint256 => uint256)) gasDeposits;
+    }
+
+    /// @notice Determines the number of slots which should be used upon
+    //          making a gas deposit.
+    uint256 public constant depositSize = 1;
+
+    /// @notice Creates a gas deposit for given address.
+    /// @dev Fills up the given number of memory slots with a non-zero value and
+    ///      consumes gas as result. That gas can be recovered during release.
+    /// @param addr Address which the deposit is tied to.
+    function depositGas(Data storage self, address addr) internal {
+        setDeposit(self, addr, 1);
+    }
+
+    /// @notice Releases a gas deposit for given address.
+    /// @dev Sets all slots which have been filled up during deposit to their
+    ///      zero values. That action causes a gas refund so the gas spent
+    ///      for deposit is recovered.
+    /// @param addr Address which the deposit is tied to.
+    function releaseGas(Data storage self, address addr) internal {
+        setDeposit(self, addr, 0);
+    }
+
+    /// @notice Sets the memory slots tied to the given address to the value
+    ///         passed as function argument.
+    /// @param addr Address which the deposit is tied to.
+    /// @param value Value which should be inserted to the memory slots.
+    function setDeposit(
+        Data storage self,
+        address addr,
+        uint256 value
+    ) internal {
+        for (uint256 i = 0; i < depositSize; i++) {
```

Comment:
`depositSize` is set to 1 which makes this for loop unnecessary. But I take that `depositSize = 1` can be changed later and that's why you added a for loop? Do you think this param should be modifiable in case gas increases for operators removal from the sortition pool?

Reply:
That's the original code from the gas station we used to have in the sortition pool (removed recently). I guess we may be interested with reserving more than one slot (see https://github.com/keep-network/keep-core/pull/2671#discussion_r741280350) so this is why this loop is here. Ideally, we would make this param governable but it's not that simple because we would need to handle existing deposits that would still use the old slot value.

Reply:
Yea, makes sense, I would just add a comment saying that this one slot should be adjusted based on the real sortition pool or system test.

Reply:
See https://github.com/keep-network/keep-core/pull/2671/commits/210861513bf978a98bd737e3b90a27ca838f0d02